delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/IdResolution.scala

```diff
@@ -0,0 +1,137 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch
+
+import ch.epfl.bluebrain.nexus.delta.kernel.search.Pagination.FromPagination
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.IdResolutionResponse._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ResourcesSearchParams
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.query.DefaultSearchRequest.RootSearch
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.query.ElasticSearchQueryError.AuthorizationFailed
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.query.{DefaultViewsQuery, ElasticSearchQueryError}
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.RdfError
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.contexts
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.api.{JsonLdApi, JsonLdOptions}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.{CompactedJsonLd, ExpandedJsonLd}
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdContent
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.searchResultsJsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.{SearchResults, SortList}
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{Label, ProjectRef, ResourceRef}
+import io.circe.JsonObject
+import monix.bio.{IO, UIO}
+
+import java.rmi.UnexpectedException
+
+/**
+  * @param defaultViewsQuery
+  *   how to list resources from the default elasticsearch views
+  * @param fetchResource
+  *   how to fetch a resource given a resourceRef and the project it lives in
+  */
+class IdResolution(
+    defaultViewsQuery: DefaultViewsQuery.Elasticsearch,
+    fetchResource: (ResourceRef, ProjectRef) => UIO[Option[JsonLdContent[_, _]]]
```

Comment:
If there is not one already, we could have a alias type for this one ?

Reply:
For MultiFetch there is MultiFetchRequest which I didn't want to use for id resolution because I thought it was confusing. Should we try to unify it or just have a type alias for this one?